pkgs/development/tools/misc/binutils/default.nix

```diff
@@ -126,7 +125,16 @@ stdenv.mkDerivation {
     # RUNPATH instead of RPATH on binaries.  This is important because
     # RUNPATH can be overriden using LD_LIBRARY_PATH at runtime.
     "--enable-new-dtags"
-  ] ++ lib.optionals gold [ "--enable-gold" "--enable-plugins" ];
+  ] ++ lib.optionals gold [
+    "--enable-gold"
+    "--enable-plugins"
+  ] ++ lib.optionals (stdenv.hostPlatform != stdenv.targetPlatform) [
+    # force target prefix if we are cross-compiling since binutils
+    # doesn't set the prefix if stdenv.hostPlatform.config and
+    # stdenv.targetPlatform.config are the same which may happen
+    # even though we are cross compiling (e. g. pkgsCross.musl64).
+    "--program-prefix=${targetPrefix}"
+  ];
```

Comment:
```suggestion
    # force target prefix. Some versions of binutils will make it empty
    # if `--host` and `--target` are too close, even if Nixpkgs thinks the
    # platforms different (e.g. because not all the info makes the `config`).
    # Other versions of binutils will always prefix if `--target` is passed, even
    # if `--host` and `--target`are the same. The easiest thing for us to do
    # is not leave it to chance, and force the program prefix to be what we
    # want it to be.
    "--program-prefix=${targetPrefix}"
  ] ++ lib.optionals gold [
    "--enable-gold"
    "--enable-plugins"
  ];
```

Reply:
Done, thank you!